devpack/src/main/java/io/neow3j/devpack/compiler/Compiler.java

```diff
@@ -2,224 +2,565 @@
 
-    public static void main(String[] args) throws Exception {
-        ClassReader reader = new ClassReader("io.neow3j.devpack.template.HelloWorldContract");
+    private static final Logger log = LoggerFactory.getLogger(Compiler.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public static final int MAX_PARAMS_COUNT = 255;
+    public static final int MAX_LOCAL_VARIABLES_COUNT = 255;
+//    public static final int MAX_STATIC_FIELDS_COUNT = 255;
+
+    private int currentNeoAddr;
+
+    /**
+     * Compiles the class with the given name to NeoVM code.
+     *
+     * @param name the fully qualified name of the class.
+     */
+    public CompilationResult compileClass(String name) throws IOException {
+        ClassReader reader = new ClassReader(name);
         ClassNode n = new ClassNode();
         reader.accept(n, 0);
+        MethodNode entryPoint = getEntryPoint(n);
+        NeoMethod neoMethod = handleMethod(entryPoint);
+        byte[] script = neoMethod.toByteArray();
+        NefFile nef = new NefFile("neow3j", new Version(3, 0, 0, 0), script);
+        return new CompilationResult(nef, buildManifest(n, nef.getScriptHash()));
+    }
+
+    private MethodNode getEntryPoint(ClassNode n) {
         MethodNode[] entryPoints = n.methods.stream()
-                .filter(m -> m.invisibleAnnotations != null &&
-                        m.invisibleAnnotations.stream()
-                                .anyMatch(a -> a.desc.equals(Type.getDescriptor(EntryPoint.class))))
+                .filter(m -> m.invisibleAnnotations != null && m.invisibleAnnotations.stream()
+                        .anyMatch(a -> a.desc.equals(Type.getDescriptor(EntryPoint.class))))
                 .toArray(MethodNode[]::new);
         if (entryPoints.length > 1) {
             throw new CompilerException("Multiple entry points found.");
         } else if (entryPoints.length == 0) {
             throw new CompilerException("No entry point found.");
         }
-        MethodNode entryPoint = entryPoints[0];
-        Iterator<AbstractInsnNode> it = entryPoint.instructions.iterator();
-        ByteArrayOutputStream ms = new ByteArrayOutputStream();
-        BinaryWriter writer = new BinaryWriter(ms);
-        while (it.hasNext()) {
-            AbstractInsnNode insn = it.next();
+        return entryPoints[0];
+    }
+
+    private NeoMethod handleMethod(MethodNode asmMethod) throws IOException {
+        NeoMethod neoMethod = new NeoMethod();
+        collectLocalVariables(asmMethod, neoMethod);
+        addMethodBeginCode(asmMethod, neoMethod);
+        for (int insnAddr = 0; insnAddr < asmMethod.instructions.size(); insnAddr++) {
+            AbstractInsnNode insn = asmMethod.instructions.get(insnAddr);
             JVMOpcode opcode = JVMOpcode.get(insn.getOpcode());
             if (opcode == null) {
                 continue;
             }
+            log.info(opcode.toString());
             switch (opcode) {
```

Comment:
I believe this switch statement contains many opcodes that should be treated the same way... like the `ICONST` ones.... that you should treat by having a "addPushNumber()" and then break it.

Assuming that, in order to improve code understandability/readability, can you add comment blocks for it? Maybe that's useful when trying to understand the whole thing.

If you think that's useless I don't strongly oppose it. Just a suggestion. 😄